src/lib/fauna-import-writer.js

```diff
@@ -160,8 +172,20 @@ input file '${inputFile}' failed to persist in Fauna due to: '${subMessage}' - C
 
     const processItems = async () => {
       // writeData has side effect of clearing out items and itemNumbers
-      await logSettlements(await writeData(items, itemNumbers))
+      const [writeOps, numRequests] = await processSettlements(
+        await writeData(items, itemNumbers)
+      )
       await waitForRateLimiter(rateLimiterBytes, dataSize, bytesPerSecondLimit)
+      await waitForRateLimiter(
+        rateLimiterWriteOps,
+        writeOps,
+        writeOpsPerSecondLimit
+      )
+      await waitForRateLimiter(
+        rateLimiterRequests,
+        numRequests,
+        requestsPerSecondLimit
+      )
```

Comment:
we need to ask the rate limiter for space to issue requests _before_ making htem; - for the write ops is the only one that should come after.

if you want to work on this together let's huddle up.